src/order/symm_diff.lean

```diff
@@ -0,0 +1,99 @@
+
+## Main declarations
+
+* `symm_diff`: the symmetric difference operator, defined as `(A \ B) ⊔ (B \ A)`
+
+In Boolean algebras, the symmetric difference operator is:
+
+* `symm_diff_comm`: commutative, and
+* `symm_diff_assoc`: associative.
+
+## Notations
+
+* `a Δ b`: `symm_diff a b`
+
+## Tags
+boolean ring, boolean algebra, symmetric differences
+-/
+
+/-- The symmetric difference operator of a structure with `⊔` and `\` is `(A \ B) ⊔ (B \ A)`. -/
+def symm_diff {α : Type*} [has_sup α] [has_sdiff α] (A B : α) : α := (A \ B) ⊔ (B \ A)
+
+infix ` Δ `:100 := symm_diff
+
+section generalized_boolean_algebra
+variables {α : Type*} [generalized_boolean_algebra α] (a b c : α)
+
+lemma symm_diff_comm : a Δ b = b Δ a := by simp only [(Δ), sup_comm]
+
+lemma sdiff_symm_diff : c \ (a Δ b) = (c ⊓ a ⊓ b) ⊔ ((c \ a) ⊓ (c \ b)) :=
+by simp only [(Δ), sdiff_sdiff_sup_sdiff']
+-- by simp only [symm_diff_eq, compl_sup, compl_inf, compl_compl]
+      -- ... = (((c \ a) ⊓ (c \ b)) ⊔ (b ⊓ (c \ b))) ⊔ (((c \ a) ⊔ b) ⊓ a) :
+      --                                                        by rw [inf_sup_left, inf_sup_right]
+      -- ... = ((c \ a) ⊓ (c \ b)) ⊔ (((c \ a) ⊔ b) ⊓ a)       : by rw [inf_sdiff_same, sup_bot_eq]
+      -- ... = ((c \ a) ⊓ (c \ b)) ⊔ (((c \ a) ⊓ a) ⊔ (b ⊓ a)) : by rw [inf_sup_right]
+      -- ... = ((c \ a) ⊓ (c \ b)) ⊔ (b ⊓ a)                   : by rw [sdiff_inf_same, bot_sup_eq]
+      -- ... = (a ⊓ b) ⊔ ((c \ a) ⊓ (c \ b))                  : by rw [@inf_comm _ _ b a, sup_comm]
+
+end generalized_boolean_algebra
+
+section boolean_algebra
+variables {α : Type*} [boolean_algebra α] (a b c : α)
+
+lemma symm_diff_eq : a Δ b = (a ⊓ bᶜ) ⊔ (b ⊓ aᶜ) := by simp only [(Δ), sdiff_eq]
+
+lemma compl_symm_diff : (a Δ b)ᶜ = (a ⊓ b) ⊔ (aᶜ ⊓ bᶜ) :=
+by simp only [←top_sdiff, sdiff_symm_diff, top_inf_eq]
+
+lemma symm_diff_symm_diff :
+  a Δ (b Δ c) = (a ⊓ b ⊓ c) ⊔ (a ⊓ bᶜ ⊓ cᶜ) ⊔ (aᶜ ⊓ b ⊓ cᶜ) ⊔ (aᶜ ⊓ bᶜ ⊓ c) :=
+calc a Δ (b Δ c) = (a ⊓ ((b ⊓ c) ⊔ (bᶜ ⊓ cᶜ))) ⊔
+                     (((b ⊓ cᶜ) ⊔ (c ⊓ bᶜ)) ⊓ aᶜ)  : by rw [symm_diff_eq, compl_symm_diff,
+                                                            symm_diff_eq]
+             ... = (a ⊓ b ⊓ c) ⊔ (a ⊓ bᶜ ⊓ cᶜ) ⊔
+                     (b ⊓ cᶜ ⊓ aᶜ) ⊔ (c ⊓ bᶜ ⊓ aᶜ) : by rw [inf_sup_left, inf_sup_right,
+                                                            ←sup_assoc, ←inf_assoc, ←inf_assoc]
+             ... = (a ⊓ b ⊓ c) ⊔ (a ⊓ bᶜ ⊓ cᶜ) ⊔
+                     (aᶜ ⊓ b ⊓ cᶜ) ⊔ (aᶜ ⊓ bᶜ ⊓ c) : begin
+                                                       congr' 1,
+                                                       { congr' 1,
+                                                         rw [inf_comm, inf_assoc], },
+                                                       { apply inf_left_right_swap }
+                                                     end
+
+lemma symm_diff_assoc : a Δ b Δ c = a Δ (b Δ c) :=
```

Comment:
This is proved here in the `boolean_algebra` case, but should be true for `generalized_boolean_algebra` as well. (All the properties of complements used in the proof could be translated into corresponding properties of complements relative to `a ⊔ b ⊔ c`, for example.)

Reply:
Agreed! I was going to move these results to `generalized_boolean_algebra` after #6775 is merged.